srcpkgs/intel-parallel-studio-xe/template

```diff
@@ -0,0 +1,591 @@
+
+repository=nonfree
+restricted=yes
+nocross=yes
+nodebug=yes
+nopie=yes
+nostrip=yes
+
+lib32disabled=yes
+
+makedepends="bsdtar sed"
+depends="bash gcc hicolor-icon-theme desktop-file-utils"
+
+if [ "$XBPS_MACHINE" = "x86_64" ]; then
+	_i_arch='intel64'
+	_bin_dir='bin64'
+	_not_arch='ia32'
+else
+	_i_arch='ia32'
+	_bin_dir='bin32'
+	_not_arch='intel64'
+fi
+
+extract_rpm() {
+	# filter out rpms for incompatible arch
+	local extract=true
+	if [[ "$1" =~ .noarch.rpm$ ]]; then
+		if [[ "$1" =~ -doc ]]; then
+			if $_remove_docs ; then
+				# don't skip doc rpms because they contain man pages
+				#extract=false
+				extract=true
+			fi
+		fi
+	else
+		if [[ "$1" =~ (-32bit.+\.x86_64|\.i486)\.rpm$ ]]; then
+			if [[ "$XBPS_MACHINE" != 'i686' ]]; then
+				extract=false
+			fi
+		else
+			if [[ "$XBPS_MACHINE" == 'i686' ]]; then
+				extract=false
+			fi
+		fi
+	fi
+	if $extract ; then
+		echo "  ${1##*/}"
+		bsdtar -xf $1 -C $2
+	else
+		echo "skipped ${1##*/}"
+	fi
+}
+
+extract_rpms() {
+	# select binary files of specific arch
+	if [ -z "$3" ]; then
+		_rpm_dir="${rpm_dir}"
+	else
+		_rpm_dir="$3"
+	fi
+	for rpm_file in ${_rpm_dir}/$1 ; do
+		extract_rpm ${rpm_file} $2
+	done
+}
+
+set_build_vars() {
+	_composer_xe_dir="compilers_and_libraries_${_year}.${_v_a}.${_v_b}"
+	rpm_dir=${wrksrc}/rpm
+	xe_build_dir=${wrksrc}/cxe_build
+	base_dir=$(realpath ${wrksrc}/..)
+	_man_dir=${xe_build_dir}/usr/share/man/man1
+}
+
+do_install() {
+	set_build_vars
+
+	# clean build directories
+	[ -d ${xe_build_dir} ] && rm -fr ${xe_build_dir}
+	mkdir -p ${xe_build_dir}
+
+	echo -e "-----------------------------------------------------------------------------------"
+	echo -e "\e[1mIMPORTANT - READ BEFORE COPYING, INSTALLING, OR USING.\e[0m"
+	echo -e ""
+	echo -e "Do not copy, install, or use the \"Materials\" provided under this license agreement (\"Agreement\"), until you"
+	echo -e "have carefully read the following terms and conditions."
+	echo -e ""
+	echo -e "By copying, installing, or otherwise using the Materials, you agree to be bound by the terms of this"
+	echo -e "Agreement. If you do not agree to the terms of this Agreement, do not copy, install, or use the Materials."
+	echo -e " - A copy of the EULA has been copied in the package directory; plase read carefully the terms and "
+	echo -e " - conditions of the Intel license before installing the packages. "
+	echo -e "-----------------------------------------------------------------------------------"
+
+	extract_rpms 'intel-parallel-studio-*.rpm' $xe_build_dir
+	extract_rpms 'intel-psxe-*.rpm' $xe_build_dir
+
+	vlicense ${wrksrc}/license.txt
+
+	mv $xe_build_dir/opt ${PKGDESTDIR}
+}
+
+intel-common-libs_package() {
+	short_desc="Intel Common Libraries $_icc_ver"
+	shlib_provides="libintlc.so.5 libcilkrts.so.5 libpdbx.so.5 libimf.so libsvml.so libirng.so libchkp.so"
```

Comment:
is it necesary?

Reply:
It seems so. I am having a lot of problems with auto-detection of shared libraries that are exported verse used by packages. It wouldn't be a concern if the whole thing were a single package, but is a problem with this being many packages in one.